WordPress/gutenberg
# Do not provide a default value for the link color property

In https://github.com/WordPress/gutenberg/pull/30541 we moved from providing a fallback to the link color such as:

```css
a {
  color:var(--wp--style--color--link, #00e);
}
```

to directly set the CSS Custom Property:

```css
body {
  --wp--style--color--link: #00e;
}
a {
  color:var(--wp--style--color--link);
}
```

This is: `--wp--style--color--link` moved from being undefined unless the theme or the user set it to always being defined. While this works fine for themes with theme.json support, some themes without theme.json may use the `--wp--style--color--link` in this way:

```css
a {
  var(--wp--style--color--link, <theme_value>);
}
```

It'll be used if it has a value and fallbacks to the theme color if it doesn't. This is what TwentyTwentyOne does.

## What this PR does

It reverts back that change so the link color CSS Custom Property is undefined if the theme or user didn't set it. It provides the fallback value as before.

```css
a {
  color:var(--wp--style--color--link, #00e);
}
```

## How to reproduce the bug

- Install TwentyTwentyOne.
- Activate Gutenberg 10.6
- Visit a post that contains links

The expected behavior is that the link color is `#28303d`, which is the value `--global--color--primary` resolves to. Instead, it is the default link color.

## How to test

Follow the instructions above and verify that it works as expected.

Test with a theme with theme.json and verify that link color still works (set the link color via `styles.elements.link.text` or as a user via the global styles sidebar in edit site).
